app/selectors/autocomplete.js

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {General} from 'mattermost-redux/constants';
+import {getMyChannels, getOtherChannels} from 'mattermost-redux/selectors/entities/channels';
+import {
+    getCurrentUser, getProfilesInCurrentChannel,
+    getProfilesNotInCurrentChannel, getProfilesInCurrentTeam
+} from 'mattermost-redux/selectors/entities/users';
+import {sortChannelsByDisplayName} from 'mattermost-redux/utils/channel_utils';
+import {sortByUsername} from 'mattermost-redux/utils/user_utils';
+
+import {getCurrentUserLocale} from 'app/selectors/i18n';
+
+export const filterMembersInChannel = createSelector(
+    getProfilesInCurrentChannel,
+    (state, opts) => opts,
+    (profilesInChannel, opts) => {
```

Comment:
A bit more detail from my previous comments, since `opts` changes whenever mapStateToProps is called for those components, this code is called each time and the memoized result is not used.